app/src/main/java/org/mozilla/fenix/library/downloads/viewholders/DownloadsListItemViewHolder.kt

```diff
@@ -24,24 +28,78 @@ class DownloadsListItemViewHolder(
 
     private var item: DownloadItem? = null
 
+    init {
+        setupMenu()
+
+        itemView.delete_downloads_button.setOnClickListener {
+            val selected = selectionHolder.selectedItems
+            if (selected.isEmpty()) {
+                downloadInteractor.onDeleteAll()
+            } else {
+                downloadInteractor.onDeleteSome(selected)
+            }
+        }
+    }
+
     fun bind(
-        item: DownloadItem
+        item: DownloadItem,
+        showDeleteButton: Boolean,
+        mode: DownloadFragmentState.Mode,
+        isPendingDeletion: Boolean = false
     ) {
-        itemView.download_layout.visibility = View.VISIBLE
+        if (isPendingDeletion) {
+            itemView.download_layout.visibility = View.GONE
+        } else {
+            itemView.download_layout.visibility = View.VISIBLE
+        }
```

Comment:
Nit: 

```
itemView.download_layout.visibility = if (isPendingDeletion) {
                View.GONE
        } else {
                View.VISIBLE
        }
```

Reply:
Maybe we can use [view.isVisible](https://android.github.io/android-ktx/core-ktx/androidx.view/android.view.-view/is-visible.html), for example `itemView.download_layout.isVisible = !isPendingDeletion `  :)

Reply:
@kglazko this is pending :)